lib/Menu.php

```diff
@@ -95,10 +94,148 @@ class Menu extends CoreEntity {
 	 *                            included in the menu. Default `0`, which is all levels.
 	 * @return \Timber\Menu
 	 */
-	public static function build(WP_Term $wp_term, array $args = []) : Menu {
-		$term = new static($wp_term->term_id, $args);
-		$term->init($wp_term);
-		return $term;
+	public static function build(?WP_Term $menu, $args = []) : ?self {
+		/**
+		 * Default arguments from wp_nav_menu() function.
+		 *
+		 * @see wp_nav_menu()
+		 */
+		$defaults = array(
+			'menu'                 => '',
+			'container'            => 'div',
+			'container_class'      => '',
+			'container_id'         => '',
+			'container_aria_label' => '',
+			'menu_class'           => 'menu',
+			'menu_id'              => '',
+			'echo'                 => true,
+			'fallback_cb'          => 'wp_page_menu',
+			'before'               => '',
+			'after'                => '',
+			'link_before'          => '',
+			'link_after'           => '',
+			'items_wrap'           => '<ul id="%1$s" class="%2$s">%3$s</ul>',
+			'item_spacing'         => 'preserve',
+			'depth'                => 0,
+			'walker'               => '',
+			'theme_location'       => '',
+		);
+
+		$args = wp_parse_args( $args, $defaults );
+
+		if ( ! in_array( $args['item_spacing'], array( 'preserve', 'discard' ), true ) ) {
+			// Invalid value, fall back to default.
+			$args['item_spacing'] = $defaults['item_spacing'];
+		}
+
+		/**
+		 * @see wp_nav_menu()
+		 */
+		$args = apply_filters( 'wp_nav_menu_args', $args );
+		$args = (object) $args;
+
+		/**
+		 * Since Timber doesn't use HTML here, try to unserialize the maybe cached menu object
+		 *
+		 * @see wp_nav_menu()
+		 */
+		$nav_menu = apply_filters( 'pre_wp_nav_menu', null, $args );
+		if( null !== $nav_menu ) {
+			try {
+				$nav_menu = unserialize($nav_menu);
+				if( $nav_menu instanceof Menu ) {
+					return $nav_menu;
+				}
+			} catch(\Throwable $e) {
+			}
+		}
+
+		// No menu term provided, get the pages menu
+		if(!$menu) {
+
+			$defaults = [
+				'sort_column'  => 'menu_order, post_title',
+				'menu_id'      => '',
+				'menu_class'   => 'menu',
+				'container'    => 'div',
+				'echo'         => true,
+				'link_before'  => '',
+				'link_after'   => '',
+				'before'       => '<ul>',
+				'after'        => '</ul>',
+				'item_spacing' => 'discard',
+				'walker'       => '',
+			];
+
+			$args = wp_parse_args( (array) $args, $defaults );
+
+			/**
+			 * @see wp_page_menu()
+			 */
+			$args = apply_filters( 'wp_page_menu_args', $args );
+
+			$nav_menu = new static(null, $args);
+			$nav_menu->init_pages_menu();
+
+			/**
+			 * @see wp_page_menu()
+			 */
+			$_nav_menu = apply_filters( 'wp_page_menu', serialize($nav_menu), $args );
```

Comment:
```suggestion
		/**
		 * Since Timber doesn’t use HTML, serialize the menu object to provide a cacheable string.
		 *
		 * Certain caching plugins will use this filter to cache a menu and return it early in the
		 * `pre_wp_nav_menu` filter.
		 *
		 * We can’t use the result of this filter, because it would return a string. That’s why we
		 * don’t assign the result of the filter to a variable.
		 *
		 * @see wp_nav_menu()
		 */
		apply_filters( 'wp_nav_menu', serialize( $nav_menu ), $args );
```